docs/how-to/hip_porting_guide.rst

```diff
@@ -707,9 +1066,226 @@ architectures with multiple execution units per multiprocessor. For example, the
+          return 1;
+      }
+
+      // Call the hipInit function using the obtained address
+      res = hipInitFunc(0);
+      if (res == hipSuccess) {
+          std::cout << "HIP runtime initialized successfully using hipGetProcAddress()." << std::endl;
+      } else {
+          std::cerr << "Failed to initialize HIP runtime using hipGetProcAddress()." << std::endl;
+      }
+
+      return 0;
+  }
+
+Per-thread default stream version request
+-----------------------------------------
+
+HIP offers functionality similar to CUDA for managing streams on a per-thread
+basis. By using ``hipStreamPerThread``, each thread can independently manage its
+default stream, simplifying operations. The following example demonstrates how
+this feature enhances performance by reducing contention and improving
+efficiency.
+
+.. code-block:: cpp
+
+  #include <hip/hip_runtime.h>
+
+  #include <iostream>
+
+  int main() {
+      // Initialize the HIP runtime
+      hipError_t res = hipInit(0);
+      if (res != hipSuccess) {
+          std::cerr << "Failed to initialize HIP runtime." << std::endl;
+          return 1;
+      }
+
+      // Get the per-thread default stream
+      hipStream_t stream = hipStreamPerThread;
+
+      // Use the stream for some operation
+      // For example, allocate memory on the device
+      void* d_ptr;
+      size_t size = 1024;
+      res = hipMalloc(&d_ptr, size);
+      if (res != hipSuccess) {
+          std::cerr << "Failed to allocate memory." << std::endl;
+          return 1;
+      }
+
+      // Perform some operation using the stream
+      // For example, set memory on the device
+      res = hipMemsetAsync(d_ptr, 0, size, stream);
+      if (res != hipSuccess) {
+          std::cerr << "Failed to set memory." << std::endl;
+          return 1;
+      }
+
+      // Synchronize the stream
+      res = hipStreamSynchronize(stream);
+      if (res != hipSuccess) {
+          std::cerr << "Failed to synchronize stream." << std::endl;
+          return 1;
+      }
+
+      std::cout << "Operation completed successfully using per-thread default stream." << std::endl;
+
+      // Free the allocated memory
+      hipFree(d_ptr);
+
+      return 0;
+  }
+
+Accessing new HIP features with a newer driver
+----------------------------------------------
+
+HIP is designed to be forward compatible, allowing newer features to be utilized
+with older toolkits, provided a compatible driver is present. Feature support
+can be verified through runtime API functions and version checks. This approach
+ensures that applications can benefit from new features and improvements in the
+HIP runtime without needing to be recompiled with a newer toolkit. The function
+:cpp:func:`hipGetProcAddress` enables dynamic querying and the use of newer
+functions offered by the HIP runtime, even if the application was built with an
+older toolkit.
+
+An example is provided for a hypothetical ``foo()`` function.
```

Comment:
Not sure why this is here. It doesn't mention CUDA, why is this in the porting guide?

Reply:
I think this applies to newer CUDA drivers? Does it make sense in that context, or should I just remove it?